### DISCUSSION TITLE
HTTP API request latest value of parameter

### DISCUSSION THREAD

**Discussion by guidoSNT** [yamcs/yamcs#917] [Q&A] [answered]
Hi Yamcs team,

I've been using the HTTP API but couldn't get around a problem while trying to request the latest acquired value of a parameter. As far as I understand, to request samples of a parameter I can do the following:

```console
curl 'http://localhost:8080/api/archive/{instance}/parameters/{name*}/samples'
```
This returns a lot of points so to get only one i can do:
```console
curl 'http://localhost:8080/api/archive/{instance}/parameters/{name*}/samples?count=1'
```
Even though it gives only one value, it uses others to get an average, so it doesn't work for what I need. How should it be done?

**Comment by fqqb** [comment, marked as answer]
The primary use of the samples call is to generate a fast summary for arbitrary periods (useful as plot input).

If all you need is the latest value, you can use this instead: https://docs.yamcs.org/yamcs-http-api/processing/get-parameter-value/
Use ``realtime`` for the processor route parameter.

**Reply by guidoSNT** [reply]
Worked well, thank you!
